gemini/sample-apps/accelerating_product_innovation/app/pages_utils/utils_get_llm_response.py

```diff
@@ -0,0 +1,96 @@
+"""
+This module provides functions for interacting with Vertex AI text generation
+model (Gemini-Pro).
+
+* generate_gemini():
+    * Utilizes the Gemini-Pro model for flexible text generation.
+    * Supports customization of generation parameters.
+    * Incorporates safety settings.
+
+* parallel_generate_search_results():
+    * Employs asynchronous requests to Gemini for search result generation.
+    * Handles potential errors during communication with the model.
+"""
+
+# pylint: disable=E0401
+
+import json
+import logging
+import os
+
+import aiohttp
+from dotenv import load_dotenv
+import vertexai
+from vertexai import generative_models
+
+load_dotenv()
+
+
+PROJECT_ID = os.getenv("PROJECT_ID")
+LOCATION = os.getenv("LOCATION")
+
+vertexai.init(project=PROJECT_ID, location=LOCATION)
+logging.basicConfig(format="%(levelname)s:%(message)s", level=logging.DEBUG)
+
+
+def generate_gemini(text_prompt: str) -> str:
+    """Generates text using the Gemini-Pro model.
+
+    Args:
+        text_prompt: The text prompt to generate from.
+
+    Returns:
+        The generated text.
+    """
+    model = generative_models.GenerativeModel("gemini-pro")
+    safety_setting = generative_models.HarmBlockThreshold.BLOCK_NONE
+    harm_category = generative_models.HarmCategory
+
+    responses = model.generate_content(
+        text_prompt,
+        generation_config={
+            "max_output_tokens": 8192,
+            "temperature": 0.001,
+            "top_p": 1,
+        },
+        safety_settings={
+            harm_category.HARM_CATEGORY_HATE_SPEECH: safety_setting,
+            harm_category.HARM_CATEGORY_DANGEROUS_CONTENT: safety_setting,
+            harm_category.HARM_CATEGORY_SEXUALLY_EXPLICIT: safety_setting,
+            harm_category.HARM_CATEGORY_HARASSMENT: safety_setting,
+        },
+        stream=True,
+    )
+    final_response = ""
+    for response in responses:
+        final_response += response.text
+    logging.debug(final_response)
+    return final_response
```

Comment:
Since you're not using the streaming feature, remove stream and you won't need the loop later.

```suggestion
    )
    logging.debug(response.text)
    return response.text
```